### PR TITLE
Fixed #5303

### LINKED ISSUE #5303: [iOS][Rendering] There's no way to get additional properties from ACOAdaptiveCard
# Platform

What platform is your issue or question related to? (Delete other platforms).

- [ ] iOS

# Author or host

Host

# Version of SDK

2.4.6

# Details

We have a need to override some properties at the card level that are normally at the host config level. For example our host config file might contain:

```
    "mycompany.contentLightColor": "FFFFFFFF",
    "mycompany.contentDarkColor": "00263EFF",
```

But for some cards, we want to override those with different colors. In the rendering hooks for the UI elements we can get a reference to the `ACOAdaptiveCard` via `ACRView.card` but the problem is that `ACOAdaptiveCard` doesn't expose any of the properties that we've included at the top level of the card.

### PR DESCRIPTION
**Pull request by jwoo-msft**
## Related Issue
Fixed #5303 

## Description
Added additional properties support at the card level for Shared model & iOS

## Sample Card
```json
 {
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.3",
    "card.cornerRadius" : 50.0,
    "body": [
        {
            "type": "Container",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Test Card",
                    "wrap": true
                }
            ],
            "my.cornerRadius" : 20.0
        }
    ]
}
```
## How Verified
How you verified the fix, including one or all of the following: 
1. Added a new Unit Test to iOS & Shared models.
2. Ran existing unit tests.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5440)

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOAdaptiveCard.mm [hunk 1]
```diff
@@ -111,4 +111,24 @@ - (void)setCard:(std::shared_ptr<AdaptiveCard> const &)card
...
+    if (_adaptiveCard) {
+        Json::Value blob = _adaptiveCard->GetAdditionalProperties();
+        if (blob.empty()) {
+            return nil;
+        }
+        Json::StreamWriterBuilder streamWriterBuilder;
+        auto writer = streamWriterBuilder.newStreamWriter();
+        std::stringstream sstream;
+        writer->write(blob, &sstream);
+        delete writer;
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOAdaptiveCard.mm:125]: If you use `std::unique_ptr<>`, its lifetime is automatically managed for you :)

```suggestion
        std::unique_ptr<StreamWriter> writer(streamWriterBuilder.newStreamWriter());
        std::stringstream sstream;
        writer->write(blob, &sstream);
```

File: source/shared/cpp/ObjectModel/SharedAdaptiveCard.cpp [hunk 2]
```diff
@@ -228,6 +233,10 @@ std::shared_ptr<ParseResult> AdaptiveCard::Deserialize(const Json::Value& json,
     // Parse optional selectAction
     result->SetSelectAction(ParseUtil::GetAction(context, json, AdaptiveCardSchemaKey::SelectAction, false));
 
+    Json::Value addtionalProperties;
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/SharedAdaptiveCard.cpp:236]: typo -- should be `additionalProperties`

File: source/shared/cpp/ObjectModel/Enums.cpp [hunk 3]
```diff
@@ -128,6 +128,7 @@ namespace AdaptiveSharedNamespace
             {AdaptiveCardSchemaKey::Requires, "requires"},
             {AdaptiveCardSchemaKey::RichTextBlock, "richTextBlock"},
             {AdaptiveCardSchemaKey::Right, "right"},
+            {AdaptiveCardSchemaKey::Schema, "$schema"},
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/Enums.cpp:131]: have you verified that the existing shared model tests pass? also, we should probably have a card-level additional properties test...

**Review by paulcam206 — CHANGES_REQUESTED** [review]

File: source/shared/cpp/ObjectModel/Enums.cpp (hunk 3 above)
Comment (jwoo-msft) [reply source/shared/cpp/ObjectModel/Enums.cpp:131]: The shared model tests are passed & new tests for card level additional properties are added to  `AdditionalPropertiesTest.cpp`

**Review by paulcam206 — APPROVED** [review]

**Pull request merged**